gen/generate.go

```diff
@@ -30,10 +30,16 @@ import (
 
 	"go.uber.org/thriftrw/compile"
 	"go.uber.org/thriftrw/internal/plugin"
+	"go.uber.org/thriftrw/plugin/api"
 
 	"go.uber.org/multierr"
 )
 
+// CodeGenerator lists possible code generators for a plugin.
```

Comment:
is 'lists possible code generators' a correct description? api.SerrviceGenerator is anything that takes Service data and returns generated code files?

Reply:
I think the idea is that if we ever add another hook to plugin/api, that'll go here as a separate code generator. (ServiceGenerator is the pluggable generator for "I want to generate code for `service`s declared in the Thrift file.")

Reply:
Yep, that's the idea. We'll see if it ever happens. :)